adam/perception/perception_graph.py

```diff
@@ -313,111 +328,113 @@ def from_schema(object_schema: ObjectStructuralSchema) -> "PerceptionGraphPatter
                     )
+                elif isinstance(node, OntologyNode):
+                    perception_node_to_pattern_node[key] = IsPropertyNodePredicate(node)
                 else:
-                    raise RuntimeError(f"Don't know how to map node {node}")
-            return schema_node_to_pattern_node[key]
+                    raise RuntimeError(f"Don't know how to map node {node}, {type(node)}")
+            return perception_node_to_pattern_node[key]
 
-        def map_edge(label: Any) -> Mapping[str, EdgePredicate]:
+        def map_edge(label: Any) -> Mapping[str, "EdgePredicate"]:
             if isinstance(label, OntologyNode):
                 return {"predicate": RelationTypeIsPredicate(label)}
             elif isinstance(label, Direction):
                 return {"predicate": DirectionPredicate.exactly_matching(label)}
             else:
                 raise RuntimeError(f"Cannot map edge {label}")
 
-        # add a node for this sub-object to the graph
-        object_graph_node = map_node(object_)
-        graph.add_node(object_graph_node)
-
-        _translate_axes(graph, object_schema, object_graph_node, map_node, map_edge)
-        _translate_geon(
-            graph,
-            object_schema,
-            mapped_owner=object_graph_node,
-            map_geon=map_node,
-            map_axis=map_node,
-            map_edge=map_edge,
-        )
-
-        regions = immutableset(
-            relation.second_slot
-            for relation in object_schema.sub_object_relations
-            if isinstance(relation.second_slot, Region)
-        )
-
-        for region in regions:
-            _translate_region(graph, region, map_node=map_node, map_edge=map_edge)
-
-        # add all sub-objects to the graph
-        # and part-of edges between this object and them.
-        for sub_object in object_schema.sub_objects:
-            PerceptionGraphPattern._translate_schema(
-                sub_object, sub_object.schema, graph, schema_node_to_pattern_node
-            )
-            graph.add_edge(
-                map_node(sub_object),
-                object_graph_node,
-                predicate=RelationTypeIsPredicate(PART_OF),
-            )
-
-        # there may be additional sub-object relationships which need to be translated to edges
-        for sub_object_relation in object_schema.sub_object_relations:
-            if sub_object_relation.negated:
-                raise RuntimeError(
-                    "Did not expect a negated relation in a sub-object schema"
+        for original_node in perception_graph.nodes:
+            # Add each node
+            pattern_node = map_node(original_node)
+            pattern_graph.add_node(pattern_node)
+
+        # Once all nodes are translated, add all edges from each node
+        for original_node in perception_graph.nodes:
+            edges_from_node = perception_graph.edges(original_node, data=True)
+            for original_edge in edges_from_node:
```

Comment:
nit/style:  Integer indexing into tuples is hard to read, so prefer to iterate as

```python
for (source_node, dest_node, edge_data) in edges_from_node:
```